### PR TITLE
[feat] Adds http status codes (Resolves #280)

### LINKED ISSUE #280: Add HttpStatus codes api seen in dart html library
### **Issue:**

Dart `html` library exports `HttpStatus` abstract class that defines various http status codes. I can't find a replacement for this in the `web` package. Currently, the `HttpStatus` is part of darts internal apis so one can't directly import it in their project but this api is exported only through dart `html` and `io` libraries. As `web` is replacing the `html` library eventually, addition of this api would be quite handy. 

### **Proposal:**

Add `HttpStatus` api similar to one exported from `html` library. I believe one can just copy the api from dart sdk which should be good enough. I'm ready to put up a PR on this.

### LINKED ISSUE #281: [feat] Add helper to convert HtmlCollection to a dart list
### **Issue:** 

When working with html element childrens which are represented through the `HtmlCollection`, if someone wants to iterate through these childrens in dart, only approach is to use a  `for` loop like follow to either create a dart list which would then allow running operations like `forEach`, `map`, etc on the list or simply iterate through them to execute any functions. 

```dart
      final childrens = <Element>[];

      for (var i = 0; i < element.children.length; i++) {
        final child = element.children.item(i);

        childrens.add(child);
      }
```

This works but it becomes tedious when I've to do the same thing in multiple places. Off course, I could add an extension for this on `HtmlCollection` but it would be nice if such helper was present in `web` package itself. 

`TouchListConvert` extension on `TouchList` is an example of such helper. And the approach seems better instead of running a for loop which results in extra computations. Adding a similar helper for converting `HtmlCollection` to a dart list would be nice addition I believe. 

### **Proposal:** 

I can see this addition going two ways, 

1. Add a similar extension on `HtmlCollection` like `TouchListConvert` that would return a `HtmlCollectionWrapper` that would be a dart list.
2. Create a generic wrapper like `TouchListConvert` which could work for both `TouchListConvert` and `HtmlCollection`, and also other web facing apis like `NodeList` that represent a list of items. This one I'm not so sure abt how one would approach for this scenario.

### PR DESCRIPTION
**Pull request by rutvik110**
[feat] Adds http status codes (Resolves #280)

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by rutvik110** [comment]
@srujzs Regarding the naming of this api, I believe keeping it's name same as the existing `HttpStatus` api inside dart sdk shouldn't be a much of an problem, mainly due to following reasons. First is the backwards compatibility for users, second is the naming conflicts arising from `dart:io` shouldn't be a problem as the io and web library can't coexists.

File: lib/src/helpers/enums.dart [hunk 1]
```diff
@@ -227,3 +227,71 @@ abstract final class Device {
   static final String propertyPrefix =
       isFirefox ? 'moz' : (isIE ? 'ms' : (isOpera ? 'o' : 'webkit'));
 }
+
+/// HTTP status codes.
```
Comment (srujzs) [inline lib/src/helpers/enums.dart:231]: nit: I'd add a non-dartdoc comment here briefly mentioning that this is a copy of the `dart:_internal` version that's only exposed through `dart:html` and `dart:io`.

**Comment by srujzs** [comment]
Makes sense to keep the name. LGTM, thanks!

**Review by srujzs — APPROVED** [review]

**Comment by kevmoo** [comment]
We should bump the version and the changelog, please!

**Comment by kevmoo** [comment]
Should be `1.1.0-wip`

File: lib/src/helpers/enums.dart (hunk 1 above)
Comment (rutvik110) [reply lib/src/helpers/enums.dart:231]: Updated.

**Comment by rutvik110** [comment]
@srujzs @kevmoo New updates: 

* Bumped the version to `1.1.0-wip` and updated changelog.
* `HttpStatus` docs updated to add a comment requested by @srujzs.

**Review by srujzs — APPROVED** [review]
LGTM!

**Comment by kevmoo** [comment]
thanks @rutvik110 !!

**Comment by rutvik110** [comment]
Happy to contribute!!💙🎯

**Pull request merged**